### PR TITLE
fix the value relation reference loose layer id matching

### PR DESCRIPTION
**Pull request by signedav**
W
ith QGIS later 3.16.0

### CODE REVIEW

File: qfieldsync/core/offline_converter.py [hunk 1]
```diff
@@ -254,29 +254,29 @@ def convert(self):
...
+
+                                layer_id = None
+                                loose_layer_id = None
+                                for offline_layer in project.mapLayers().values():
+                                    if offline_layer.customProperty('remoteSource') == original_layer_info[online_layer_id][0]:
+                                        #  First try strict matching: the offline layer should have a "remoteSource" property
+                                        layer_id = offline_layer.id()
+                                        break
+                                    elif Qgis.QGIS_VERSION_INT < 31601 and offline_layer.name().startswith(original_layer_info[online_layer_id][1] + ' ') or \
+                                            Qgis.QGIS_VERSION_INT >= 31601 and offline_layer_name() == original_layer_info[online_layer_id][1]:
```
Comment (m-kuhn) [inline qfieldsync/core/offline_converter.py:273]: Is `offline_layer_name()` correct?

Comment (signedav) [reply qfieldsync/core/offline_converter.py:273]: No. Weird that I didn't run into an error. 
I'll have another look...

**Review by m-kuhn — APPROVED** [review]
Tested manually, works

**Comment by signedav** [comment]
Thanks a lot.

Patch release when merge?

**Comment by m-kuhn** [comment]
Patch release, yes!

**Pull request merged**